pkg/etcdcli/etcdcli.go

```diff
@@ -268,7 +321,7 @@ func (g *etcdClientGetter) MemberRemove(ctx context.Context, memberID uint64) er
 }
 
 func (g *etcdClientGetter) MemberList(ctx context.Context) ([]*etcdserverpb.Member, error) {
-	cli, err := g.getEtcdClient()
+	cli, err := g.getEtcdClientWithoutConfigMap()
```

Comment:
This bypasses the `etcd-endpoints` configmap for all MemberList calls which is not quite what I was intending. Only for the very specific case of when the endpoints controller has to recreate the configmap.

Since the configmap is used to add the `etcd-bootstrap` member endpoint to the client it will still be necessary early on during bootstrap.

To revise my earlier comments, I think we can make the configmap retrieval in `getEtcdClient()` a best effort, so that it is only bypassed when the configmap is actually deleted or missing.

```Go
import (
	apierrrors "k8s.io/apimachinery/pkg/api/errors"
)

...
configmap, err := g.configmapsLister.ConfigMaps(operatorclient.TargetNamespace).Get("etcd-endpoints")
if err == nil {
	if bootstrapIP, ok := configmap.Annotations[BootstrapIPAnnotationKey]; ok && bootstrapIP != "" {
		// escape if IPv6
		if net.ParseIP(bootstrapIP).To4() == nil {
			bootstrapIP = "[" + bootstrapIP + "]"
		}
		etcdEndpoints = append(etcdEndpoints, fmt.Sprintf("https://%s:2379", bootstrapIP))
	}
} else if err != nil && !apierrrors.IsNotFound(err){
	// Ignore errors if the configmap does not exist or was deleted
	return nil, fmt.Errorf("failed to list endpoints: %w", err)
}
```

That way, MemberList() and subsequently `getEtcdClient()` won't fail if the configmap is missing, but still use it otherwise. And we won't have to add a duplicate method just for that case.

Reply:
👍🏽 🙏🏽